lib/edgedb/edgeql/generator.ex

```diff
@@ -199,16 +204,19 @@ defmodule EdgeDB.EdgeQL.Generator do
       generate_query_module(
         query_file: query_file,
         module_name: module_name,
-        types: types(),
+        types: types,
         shape: rendered_shape,
         schema: rendered_schema,
-        query_function: @cardinality_to_function[query.result_cardinality],
-        should_render_type_for_shape: complex_shape?(raw_shape),
-        result_type: (complex_shape?(raw_shape) && "result()") || rendered_shape,
+        should_render_type_for_shape: rendered_schema && complex?,
```

Comment:
This code probably is wrong why handling the case where an empty object is returned from the database. The following code for `test/support/codegen/edgeql/types/insert_f_named.edgeql` will be generated for `query/3` and `query!/3`:
```elixir
  @spec query(
          client :: EdgeDB.client(),
          args :: args(),
          opts :: list(EdgeDB.query_option())
        ) ::
          {:ok, Result.t()}
          | {:error, reason}
        when reason: any()

  @spec query!(
          client :: EdgeDB.client(),
          args :: args(),
          opts :: list(EdgeDB.query_option())
        ) :: Result.t()
```

But the `Result` module itself won't be generated.  In this case, the structure should also be generated, but it will have no fields